main.nf

```diff
@@ -222,244 +252,362 @@ if (params.help){
+// Validate BAM input isn't set to paired_end
+if ( params.bam && !params.single_end ) {
+  exit 1, "[nf-core/eager] error: bams can only be specified with --single_end. Please check input command."
 }
 
 // Validate that skip_collapse is only set to True for paired_end reads!
-if (params.skip_collapse  && params.single_end){
-    exit 1, "--skip_collapse can only be set for paired_end samples!"
+if (!has_extension(params.input, "tsv") && params.skip_collapse  && params.single_end){
+    exit 1, "[nf-core/eager] error: --skip_collapse can only be set for paired_end samples."
 }
 
-// Strip mode sanity checking
-if (params.strip_input_fastq){
-    if (!(['strip','replace'].contains(params.strip_mode))) {
-        exit 1, "--strip_mode can only be set to strip or replace!"
+// Host removal mode validation
+if (params.hostremoval_input_fastq){
+    if (!(['remove','replace'].contains(params.hostremoval_mode))) {
+        exit 1, "[nf-core/eager] error: --hostremoval_mode can only be set to remove or replace."
     }
+}
 
-    if (params.bam && !params.run_convertbam) {
-        exit 1, "--strip_input_fastq can only be used on FASTQ, but you gave BAM input and didn't specify --run_convertbam!"
-    }
+if (params.bam_unmapped_type == '') {
+    exit 1, "[nf-core/eager] error: please specify valid unmapped read output format. Options: 'discard', 'keep', 'bam', 'fastq', 'both'. Found parameter: --bam_unmapped_type '${params.bam_unmapped_type}'."
 }
 
-// Mapper sanity checking
-if(params.mapper != "bwaaln" && params.mapper != "bwamem" && params.mapper != "circularmapper") {
-    exit 1, "Please specify a valid mapper. Options: 'bwaaln', 'bwamem', 'circularmapper'. You gave: ${params.mapper}!"
+// Bedtools validation
+if(params.run_bedtools_coverage && params.anno_file == ''){
+  exit 1, "[nf-core/eager] error: you have turned on bedtools coverage, but not specified a BED or GFF file with --anno_file. Please validate your parameters."
 }
 
-if (params.bam_discard_unmapped && params.bam_unmapped_type == '') {
-    exit 1, "Please specify valid unmapped read output format. Options: 'discard', 'bam', 'fastq', 'both'!"
+// Set up channels for annotation file
+if (!params.run_bedtools_coverage){
+  ch_anno_for_bedtools = Channel.empty()
+} else {
+  Channel
+    ch_anno_for_bedtools = Channel.fromPath(params.anno_file, checkIfExists: true)
+    .ifEmpty { exit 1, "[nf-core/eager] error: bedtools annotation file not found. Supplied parameter: --anno_file ${params.anno_file}."}
 }
 
-// Bedtools sanity checking
-if(params.run_bedtools_coverage && params.anno_file == ''){
-  exit 1, "You have turned on bedtools coverage, but not specified a BED or GFF file with --anno_file. Please validate your parameters!"
+// BAM filtering validation
+if (!params.run_bam_filtering && params.bam_mapping_quality_threshold != 0) {
+  exit 1, "[nf-core/eager] error: please turn on BAM filtering if you want to perform mapping quality filtering! Give --run_bam_filtering."
```

Comment:
```suggestion
  exit 1, "[nf-core/eager] error: please turn on BAM filtering if you want to perform mapping quality filtering! Give `--run_bam_filtering`."
```

Reply:
I prefer not to do add the back-ticks in the error messages as these are printed to the terminal where backticks have a different 'meaning'. However to make the the suggestion more clear, I've replaced it as: `Provide: --run_bam_filtering`, instead (more similar to the other ways I've done this. So wherever you commented with adding the backticks I'll have replaced it with the system above instead.